tasks/publish-pyxis-repository/publish-pyxis-repository.yaml

```
@@ -136,6 +144,24 @@ spec:
 
             curl --retry 5 --key /tmp/key --cert /tmp/crt "${PYXIS_URL}/v1/repositories/id/${PYXIS_REPOSITORY_ID}" \
                 -X PATCH -H 'Content-Type: application/json' --data-binary "${PAYLOAD}"
+
+            # Determine the correct CATALOG_BASE_URL based on the repository prefix
+            if [[ "$REPOSITORY" == quay.io/redhat-prod/* ]]
+            then
+              CATALOG_BASE_URL="https://catalog.redhat.com/software/containers"
+            elif [[ "$REPOSITORY" == quay.io/redhat-pending/* ]]
+            then
+              CATALOG_BASE_URL="https://catalog.stage.redhat.com/software/containers"
+            else
+              echo "Unknown repository prefix. Exiting..."
+              exit 1
+            fi
+
+            URL="${CATALOG_BASE_URL}/${PYXIS_REPOSITORY}/${PYXIS_REPOSITORY_ID}"
+            RESULTS_JSON=$(jq --arg name "$COMPONENT_NAME" --arg url "$URL" \
+              '.catalog_urls += [{"name": $name, "url": $url}]' <<< "$RESULTS_JSON")
         done
 
+        echo -n "${RESULTS_JSON}" | tee "$RESULTS_FILE"
```

Why the `-n` here? It's completely ok to have a new line at the end here. I would probably go even further and format the json again, so that it's more readable in the logs:
```suggestion
        jq <<< "${RESULTS_JSON}" | tee "$RESULTS_FILE"
```

---

Make sense, applied suggested changes. Thanks!